lib/model/model.go

```diff
@@ -2657,11 +2658,14 @@ func newFileInfoBatch(fn func([]protocol.FileInfo) error) *fileInfoBatch {
 
 func (b *fileInfoBatch) append(f protocol.FileInfo) {
 	b.infos = append(b.infos, f)
-	b.size += f.ProtoSize()
+	if !f.IsDeleted() {
```

Comment:
This means no flushing on delete detection and thus huge batches when a lot is deleted at once. And I don't think it's necessary anyway: If you just don't call `flushIfFull` between adding the newly added file and checking/adding a corresponding deleted file you are fine too.

Reply:
The index sender uses this and it doesn't know when to stop.

Actually I don't think this actually does what I intended.

You could still add something that is not deleted, that drags it past the full threshold forces it to flush, and then the delete is the first thing in the next batch which could have been used to detect a rename.

We probably need to change the api to something like:
```
if batch.wouldFillBatch(fi) {
   batch.flush()
}
batch.add(fi)
```

then when full, if "next" item is not a delete, I say I would be full, and force that to the next batch.
if "next" item is a delete, I check that previous item is not a delete, if it's not a delete, I accept the last delete, if previous item is a delete, I say I am already full.

Reply:
I think I addressed this neatly.